pkg/testcase/pod.go

```diff
@@ -36,14 +38,25 @@ func processPodStatus(
 	pod shared.Pod,
 	podAssertRestarts, podAssertReady, podAssertStatus assert.PodAssertFunc,
 ) {
+	cluster := factory.ClusterConfig(GinkgoT())
+
+	// process Helm install status that should be completed.
 	if strings.Contains(pod.Name, "helm-install") {
 		g.Expect(pod.Status).Should(Equal(statusCompleted), pod.Name)
+
+		// process system-upgrade apply status thats should be completed or errors bellow.
 	} else if strings.Contains(pod.Name, "apply") && strings.Contains(pod.NameSpace, "system-upgrade") {
 		g.Expect(pod.Status).Should(SatisfyAny(
 			ContainSubstring("Unknown"),
 			ContainSubstring("Init:Error"),
 			Equal(statusCompleted),
 		), pod.Name)
+
+		// process cilium-operator status that should be at least one running pod.
+	} else if strings.Contains(pod.Name, "cilium-operator") && cluster.Config.Product == "rke2" && cluster.NumServers == 1 {
```

Comment:
This should probably be the aggregate of the number of servers and workers I think, since it'll have both pods running in the case of 1 server 1 worker or just 2 servers

Reply:
So this check wont be triggered if the server_count is more than 1 even with 0 workers?

Reply:
Also I think it will be simpler to check `cni: cilium` to  be present in `server_flags` tf var, just a suggestion

Reply:
> So this check wont be triggered if the server_count is more than 1 even with 0 workers?


this check its going to be triggered if:

- product  rke2 + has cilium operator pod + it has only one server +  ( 0 agents,  what i mentioned above )

Reply:
> Also I think it will be simpler to check cni: cilium to be present in server_flags tf var, just a suggestion

but i will need to also check for servers and agents, so since we are here validating pods and we already have a pod structure available, i choose  to be more specific.

Reply:
Got it, I wanted to understand if we should always do this check regardless of the server/agent count and be consistent to check the CNI specific pods across other CNI's as well

Reply:
no need, only when 1 server alone